web/static/js/explore.js

```diff
@@ -252,6 +252,74 @@ $.fn.dataTable.Api.register( 'clearPipeline()', function () {
   } );
 } );
 
+
+/**
+ * Function fetches log data based on params specified by user and download data in specified fileType
+ * @param {*} api_endpoint : API endpoint URL
+ * @param {*} column_list : List of Columns for the selected event type
+ * @param {*} api_params  : GET parameters for the API call
+ * @param {*} fileType    : File Type ( Json, Excel, Csv)
+ */
+function downloadLogData(api_endpoint, column_list, api_params, fileType) {
+    $.ajax({
+        type: "GET",
+        url: api_endpoint,
+        data: {
+            ...api_params,
+            limit: 9999
+        },
+        success: function (result, status, xhr) {
+            const data = result.data
+            console.log(data.length)
+            switch (fileType) {
+                case "JSON": {
+                    const filename = get_export_fileName('json') + ".json"
+                    const dataStr = "data:text/json;charset=utf-8," + encodeURIComponent(JSON.stringify(data, undefined, 2));
+                    const downloadAnchorNode = document.createElement('a');
+                    downloadAnchorNode.setAttribute("href", dataStr);
+                    downloadAnchorNode.setAttribute("download", filename);
+                    document.body.appendChild(downloadAnchorNode);
+                    downloadAnchorNode.click();
+                    downloadAnchorNode.remove();
+                    break
+                }
+                case "CSV": {
+                    let csvFileData = "";
+                    for (let column in column_list) {
+                        csvFileData += `"${column_list[column].data}"` + ","
+                    }
+                    csvFileData += "\n"
+                    for (let index = 0; index < data.length; index++) {
+                        for (let column in column_list) {
+                            csvFileData += `"${data[index][column_list[column].data]}"` + ","
+                        }
+                        csvFileData += "\n"
+                    }
+                    const filename = get_export_fileName('csv') + "csv"
+
+                    const dataStr = "data:text/csv;charset=utf-8," + encodeURIComponent(csvFileData);
+                    const downloadAnchorNode = document.createElement('a');
+                    downloadAnchorNode.setAttribute("href", dataStr);
+                    downloadAnchorNode.setAttribute("download", filename);
+                    document.body.appendChild(downloadAnchorNode);
+                    downloadAnchorNode.click();
+                    downloadAnchorNode.remove();
```

Comment:
is this needed? i checked few examples where remove wasn't used
if needed then let it be, am just asking out of curiosity

Reply:
ill check